Add rnaseq input

christopher-mohr opened the pull request:
This adds the option to provide expression values for the peptide prediction and includes the following changes
- adapt sample sheet check 
- handle expression file along channels and add it to `epytope_peptide_prediction`
- add functionality in `epaa.py` for parsing the expression values and mapping it to the peptide predictions.

Currently the expected format of the provided expression values is the one generated by `salmon` as given in the `rnaseq` pipeline on merged transcript/gene level.

We still need functionality from `epytope` that will come with the next release. 

Resolves #134.

<!--
# nf-core/epitopeprediction pull request

Many thanks for contributing to nf-core/epitopeprediction!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/epitopeprediction _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

jonasscheid commented on bin/epaa.py:
```diff
@@ -1206,24 +1211,26 @@ def __main__():
+    if args.expression is not None:
+        rnaseq_results = pd.read_csv(args.expression, sep='\t', header=0)
+
+        measure = "count" if "count" in args.expression else "TPM"
```
Is the name of the measure type always in the rnaseq filename? Might be also an option to check the colnames within the file since we read in the file anyways

christopher-mohr replied:
According to the `rnaseq` pipeline output I saw it is always contained in the filename. You are right, that it is not the most robust solution but I think you can't tell from the column names which type it is.

jonasscheid replied:
Alright, then lets keep it like that :)

jonasscheid commented on bin/check_samplesheet.py:
```diff
@@ -267,15 +278,20 @@ def check_samplesheet(file_in, file_out):
+                sample_info = [sample, alleles, mhcclass, expression, filename]
+            else:
+                sample, alleles, mhcclass, filename = lspl[: len(HEADER)]
+                sample_info = [sample, alleles, mhcclass, "", filename]
```
A I see you are providing an empty path using this string in the prediction module. An alternative suggestion I'll comment there

christopher-mohr replied:
The way I currently implemented it, it is easier to keep "" or `None` for further processing.

christopher-mohr replied:
Actually with `None` it will fail later when the "validated sample sheet" is written. So if you do not insist I would keep "empty string". ;)

jonasscheid replied:
Yes noticed that then as well :D

jonasscheid commented on modules/local/epytope_peptide_prediction.nf:
```diff
@@ -9,6 +9,7 @@ process EPYTOPE_PEPTIDE_PREDICTION {
     input:
     tuple val(meta), path(splitted), path(software_versions)
     val netmhc_paths
+    path(expression)
```
We could do it with the rnaseq file similar than with the optional netmhc_paths, meaning declaring it as a val instead of a path.

jonasscheid reviewed (commented):
Some suggestions regarding the optional handling of the expression column in the sample sheet

christopher-mohr commented:
Unfortunately, another `Ensembl` archive has been removed for `GRCh38`, which is the reason why the `grch38` test failed. I updated the used archive to the next available one for now. We will improve the interface to `Ensembl` biomart with the next `epytope` release for which I already rewrote the biomart adapter.

jonasscheid reviewed (approved):
Comments discussed and resolved ✅ LGTM if tests will run through now 👍

jonasscheid commented:
Closing for now because it is stale since 3 years, but keeping it in mind in #134
